src/paas_charm/charm.py

```diff
@@ -492,7 +502,7 @@ def _create_charm_state(self) -> CharmState:
             s3_connection_info=self._s3.get_s3_connection_info() if self._s3 else None,
             saml_relation_data=saml_relation_data,
             rabbitmq_uri=self._rabbitmq.rabbitmq_uri() if self._rabbitmq else None,
-            tempo_parameters=TempoParameters.from_charm(name=self.app.name, tracing=self._tracing),
+            tempo_relation_data=tempo_relation_data,
```

Comment:
Let's change this to pass the tracing requires object and the app name and figure out the rest in charm state